dali/test/python/test_operator_decoders_image.py

```diff
@@ -389,3 +389,23 @@ def pipe():
     p = pipe()
     p.build()
     p.run()
+
+def test_tiff_palette():
+    normal = os.path.join(test_data_root, good_path, "tiff", "0/cat-300572_640.tiff")
+    palette = os.path.join(test_data_root, good_path, "tiff", "0/cat-300572_640_palette.tiff")
+
+    @pipeline_def(batch_size=2, device_id=0, num_threads=1)
+    def pipe():
+        encoded, _ = fn.readers.file(files=[normal, palette])
+        peeked_shapes = fn.peek_image_shape(encoded)
+        decoded = fn.decoders.image(encoded, device='cpu')
+        return decoded, peeked_shapes
+
+    p = pipe()
+    p.build()
+    imgs, peeked_shapes = p.run()
+    assert (peeked_shapes.at(0) == peeked_shapes.at(1)).all(), \
+        "Invalid peeked shape of palette TIFF"
+
+    delta = np.abs(imgs.at(0).astype('float') - imgs.at(1).astype('float'))/256
```

Comment:
This is fine, but normally we use:
```
np.testing.assert_allclose(a, b, rtol=1e-6)
```
or so

Reply:
Here I'm comparing original image with it's palette-encoded equivalent. Palette encoding changes the colors slightly, but in some places the changes can be significant. That's why I'm asking "is 90% of the image OK?" with:
```
np.quantile(delta, 0.9) < 0.05
```
instead of asking "is whole image OK?" with `allclose`.

Reply:
Another approach to this would be re-encoding the palette image as a normal TIFF again to serve as a reference for this test. I'm not sure if cluttering DALI_extra with yet another variant of this cat image is worth it.

Reply:
OK, fair enough. It's fine as it is now